### PR TITLE
PG16 compatibility - new options to vacuum and analyze

### PR DESCRIPTION
**Pull request by naisila**
Adds buffer_usage_limit to vacuum and analyze
Adds process_main, skip_database_stats, only_database_stats to vacuum

Important Note: adding these options is actually required for check-vanilla tests to succeed. However, in concept, this PR belongs to "PG16 new features", rather than "PG16 regression tests sanity"

Relevant PG commits:
https://github.com/postgres/postgres/commit/1cbbee03385763b066ae3961fc61f2cd01a0d0d7 1cbbee03385763b066ae3961fc61f2cd01a0d0d7
https://github.com/postgres/postgres/commit/4211fbd8413b26e0abedbe4338aa7cda2cd469b4 4211fbd8413b26e0abedbe4338aa7cda2cd469b4
https://github.com/postgres/postgres/commit/a46a7011b27188af526047a111969f257aaf4db8 a46a7011b27188af526047a111969f257aaf4db8

This PR should be merged first https://github.com/citusdata/citus/pull/7122

### CODE REVIEW

File: src/backend/distributed/commands/vacuum.c [hunk 1]
```diff
@@ -296,10 +300,12 @@ DeparseVacuumStmtPrefix(CitusVacuumParams vacuumParams)
 {
 	int vacuumFlags = vacuumParams.options;
 	StringInfo vacuumPrefix = makeStringInfo();
+	bool is_vacuum = false;
```
Comment (onderkalaci) [inline src/backend/distributed/commands/vacuum.c:303]: having an additional variable to track is_vacuum doesn't feel very intuitive given in all other cases we rely on vacuumFlags.

Can we do the same when is_vacuum is used such that we check vacuumFlags & VACOPT_VACUUM? If you prefer, you can also define a macro for that

**Comment by onderkalaci** [comment]
> Important Note 1: this PR also fixes default process_toast value: in PG it is true by default, in Citus it was wrongly false.

That's not good. Do you see it possible to have this fix in a separate commit such that we can backport?

File: src/backend/distributed/commands/vacuum.c [hunk 2]
```diff
@@ -519,13 +571,39 @@ VacuumStmtParams(VacuumStmt *vacstmt)
...
+		{
+			const char *hintmsg;
+			int result;
+			char *vac_buffer_size = defGetString(opt);
+
+			/*
+			 * Check that the specified value is valid and the size falls
+			 * within the hard upper and lower limits if it is not 0.
+			 */
+			if (!parse_int(vac_buffer_size, &result, GUC_UNIT_KB, &hintmsg) ||
```
Comment (onderkalaci) [inline src/backend/distributed/commands/vacuum.c:585]: why do we try to do it here? Isn't Postgres already handing it? We execute VACUUM on the coordinator shell table first anyway, and if this is wrong, we should get the error from PG?

**Comment by naisila** [comment]
> That's not good. Do you see it possible to have this fix in a separate commit such that we can backport?

@onderkalaci Thanks for your insights, I was already in doubts about this that's why I wanted to clearly note that I am changing process_toast as well.

In that case let me open a cleaner PR for PROCESS_TOAST separately, and update this one.

File: src/backend/distributed/commands/vacuum.c (hunk 2 above)
Comment (naisila) [reply src/backend/distributed/commands/vacuum.c:585]: You are right, I don't need error checks here

File: src/backend/distributed/commands/vacuum.c (hunk 1 above)
Comment (naisila) [reply src/backend/distributed/commands/vacuum.c:303]: Let me try that one

**Review by onderkalaci — APPROVED** [review]

**Pull request merged**